features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -2467,24 +2526,23 @@ ble_error_t GenericGap::setPeriodicAdvertisingPayload(
 
     typedef pal::advertising_fragment_description_t op_t;
 
-    for (size_t i = 0, end = payload.size();
-        (i < end) || (i == 0 && end == 0);
-        i += MAX_HCI_DATA_LENGTH
-    ) {
+    const size_t hci_length = _pal_gap.get_maximum_hci_advertising_data_length();
+
+    for (size_t i = 0, end = payload.size(); (i < end) || (i == 0 && end == 0); i += hci_length) {
```

Comment:
Is there a particular reason why `if (end == 0)` doesn't come before this weird for loop header?

Reply:
not my code (incidental style change when updating hci_length) but looking at it I can't see a way of removing the conditions without having to duplicate the code (or refactor the inside into a func)

Reply:
We still have to execute one iteration of the loop if `end == 0` . Thinking about it now, there is several alternative: 
* Use a `do { } while` loop. The pattern fits the use case as the first iteration must be executed unconditionally. However it mean extracting iteration variables outside of the iteration scope 😞 .
* rewrite the `for` condition into: `i == 0 || i < end`. The `end == 0` is not necessary.